### ISSUE TITLE
juicefs hadoop sdk mkdir fail

### ISSUE THREAD

**Issue by tangyoupeng** [juicedata/juicefs#977]
```
org.apache.hadoop.hive.ql.metadata.HiveException: java.io.IOException: jfs://xxxx
        at org.apache.hadoop.hive.ql.io.HiveFileFormatUtils.getHiveRecordWriter(HiveFileFormatUtils.java:249)
        at org.apache.spark.sql.hive.execution.HiveOutputWriter.<init>(HiveFileFormat.scala:123)
        at org.apache.spark.sql.hive.execution.HiveFileFormat$$anon$1.newInstance(HiveFileFormat.scala:103)
        at org.apache.spark.sql.execution.datasources.DynamicPartitionDataWriter.newOutputWriter(FileFormatDataWriter.scala:236)
        at org.apache.spark.sql.execution.datasources.DynamicPartitionDataWriter.write(FileFormatDataWriter.scala:260)
        at org.apache.spark.sql.execution.datasources.FileFormatWriter$$anonfun$org$apache$spark$sql$execution$datasources$FileFormatWriter$$executeTask$3.apply(FileFormatWriter.scala:245)
        at org.apache.spark.sql.execution.datasources.FileFormatWriter$$anonfun$org$apache$spark$sql$execution$datasources$FileFormatWriter$$executeTask$3.apply(FileFormatWriter.scala:242)
        at org.apache.spark.util.Utils$.tryWithSafeFinallyAndFailureCallbacks(Utils.scala:1394)
        at org.apache.spark.sql.execution.datasources.FileFormatWriter$.org$apache$spark$sql$execution$datasources$FileFormatWriter$$executeTask(FileFormatWriter.scala:248)
        at org.apache.spark.sql.execution.datasources.FileFormatWriter$$anonfun$write$1.apply(FileFormatWriter.scala:170)
        at org.apache.spark.sql.execution.datasources.FileFormatWriter$$anonfun$write$1.apply(FileFormatWriter.scala:169)
        at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:90)
        at org.apache.spark.scheduler.Task.run(Task.scala:123)
        at org.apache.spark.executor.Executor$TaskRunner$$anonfun$10.apply(Executor.scala:408)
        at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1360)
        at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:414)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: jfs://xxxxx/.hive-staging_hive_2021-11-14_16-23-42_159_6170667678531491126-1/-ext-10000/_temporary/0/_temporary
        at io.juicefs.JuiceFileSystemImpl.error(JuiceFileSystemImpl.java:227)
        at io.juicefs.JuiceFileSystemImpl.mkdirs(JuiceFileSystemImpl.java:1483)
        at io.juicefs.JuiceFileSystemImpl.mkdirs(JuiceFileSystemImpl.java:1480)
        at io.juicefs.JuiceFileSystemImpl.mkdirs(JuiceFileSystemImpl.java:1480)
        at io.juicefs.JuiceFileSystemImpl.create(JuiceFileSystemImpl.java:1113)
        at org.apache.hadoop.fs.FilterFileSystem.create(FilterFileSystem.java:180)
        at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:931)
        at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:912)
        at org.apache.hadoop.fs.FileSystem.create(FileSystem.java:808)
        at parquet.hadoop.ParquetFileWriter.<init>(ParquetFileWriter.java:176)
        at parquet.hadoop.ParquetFileWriter.<init>(ParquetFileWriter.java:160)
        at parquet.hadoop.ParquetOutputFormat.getRecordWriter(ParquetOutputFormat.java:289)
        at parquet.hadoop.ParquetOutputFormat.getRecordWriter(ParquetOutputFormat.java:267)
        at org.apache.hadoop.hive.ql.io.parquet.write.ParquetRecordWriterWrapper.<init>(ParquetRecordWriterWrapper.java:65)
        at org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat.getParquerRecordWriterWrapper(MapredParquetOutputFormat.java:125)
        at org.apache.hadoop.hive.ql.io.parquet.MapredParquetOutputFormat.getHiveRecordWriter(MapredParquetOutputFormat.java:114)
        at org.apache.hadoop.hive.ql.io.HiveFileFormatUtils.getRecordWriter(HiveFileFormatUtils.java:261)
        at org.apache.hadoop.hive.ql.io.HiveFileFormatUtils.getHiveRecordWriter(HiveFileFormatUtils.java:246)
        ... 18 more
```
```
SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
2021/11/14 17:31:09.343986 juicefs[120924] <ERROR>: error: redis: transaction failed
        /go/src/runtime/debug/stack.go:24 +0x9f
github.com/juicedata/juicefs/pkg/meta.errno(0x7f4988a7afa0, 0x7f4988a44c10, 0xc001087200)
        /workspace/juicefs/pkg/meta/redis.go:853 +0xd9
github.com/juicedata/juicefs/pkg/meta.(*redisMeta).txn(0xc000a40000, 0x7f4988acd300, 0xc0010872c0, 0xc0012a9be8, 0xc0012a9bc8, 0x2, 0x2, 0x0)
        /workspace/juicefs/pkg/meta/redis.go:928 +0x3a5
github.com/juicedata/juicefs/pkg/meta.(*redisMeta).mknod(0xc000a40000, 0x7f4988acd300, 0xc0010872c0, 0x2dde50, 0xc0002e809c, 0x2b, 0x1c00002, 0x0, 0x0, 0x0, ...)
        /workspace/juicefs/pkg/meta/redis.go:1250 +0x3e5
github.com/juicedata/juicefs/pkg/meta.(*redisMeta).Mkdir(0xc000a40000, 0x7f4988acd300, 0xc0010872c0, 0x2dde50, 0xc0002e809c, 0x2b, 0x1c0, 0xc00050a958, 0x0, 0x0)
        /workspace/juicefs/pkg/meta/redis.go:1326 +0x126
github.com/juicedata/juicefs/pkg/fs.(*FileSystem).Mkdir(0xc00111e100, 0x7f4988acd300, 0xc0010872c0, 0xc0002e8000, 0xc7, 0xc0001701c0, 0x0)
        /workspace/juicefs/pkg/fs/fs.go:326 +0x3d3
main.jfs_mkdir(0x9f, 0xe1a0a0, 0x7f49a3b1ca90, 0xc0000001c0, 0x28)
        /workspace/juicefs/sdk/java/libjfs/main.go:594 +0xbc
main._cgoexpwrap_fc2224ef1e95_jfs_mkdir(0x9f, 0xe1a0a0, 0x7f49a3b1ca90, 0x1c0, 0x0)
        _cgo_gotypes.go:251 +0x49
```

**Comment by davies** [comment]
Show to reproduce this?

**Comment by davies** [comment]
This is caused by concurrent operations on the same directory, it retry 50 times in a short amount of time, and failed.

Fixed by #999
